masterbase/lib.py

```diff
@@ -336,34 +365,40 @@ def close_session_helper(engine: Engine, steam_id: str, streaming_sessions: dict
     if latest_session_id is None:
         return "User has never been in a session!"
 
-    demo_path = make_demo_path(latest_session_id)
-    demo_path_exists = os.path.exists(demo_path)
+    # find session manager and socket/key...
+    session_manager = None
+    socket = None
+    for _socket in streaming_sessions:
+        if streaming_sessions[_socket].session_id == latest_session_id:
+            session_manager = streaming_sessions[_socket]
+            socket = _socket
 
     current_time = datetime.now().astimezone(timezone.utc)
 
-    if latest_session_id is None or not demo_path_exists:
+    if session_manager is None:
         _close_session_without_demo(engine, steam_id, current_time)
         msg = "No active session found, closing anyway."
 
-    elif latest_session_id is not None and demo_path_exists:
-        _close_session_with_demo(engine, steam_id, latest_session_id, current_time, demo_path)
-        os.remove(demo_path)
+    elif session_manager is not None and os.path.exists(session_manager.demo_path):
```

Comment:
`session_manager is not None` check is redundant here. Will always be true in the case we fall through to the `elif`